services/QuillLMS/app/controllers/api/v1/activities_controller.rb

```diff
@@ -85,6 +87,29 @@ def published_edition
     render json: {}
   end
 
+  def question_health
```

Comment:
I would extract lines 93-109 into a new object, yielding something like:
```
question_health = QuestionHealth.new(question, tool).run
render json: { question_health: question_health)
```
This keeps the controller action at the same level of abstraction while also encapsulating the logic for calculating health.